test/remote/gateways/remote_hi_pay_test.rb

```diff
@@ -109,4 +109,74 @@ def test_successful_multiple_purchases_with_single_store
     assert_success response2
     assert_include 'Captured', response2.message
   end
+
+  def test_successful_refund
+    purchase_response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success purchase_response
+    assert_include 'Captured', purchase_response.message
+
+    response = @gateway.refund(@amount, purchase_response.authorization, @options)
+    assert_success response
+    assert_include 'Refund Requested', response.message
+    assert_include response.params['authorizedAmount'], '5.00'
+    assert_include response.params['capturedAmount'], '5.00'
+    assert_include response.params['refundedAmount'], '5.00'
+    assert_equal purchase_response.authorization, response.authorization
+  end
+
+  def test_successful_partial_capture_refund
+    authorize_response = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success authorize_response
+    assert_include 'Authorized', authorize_response.message
+    assert_include authorize_response.params['authorizedAmount'], '5.00'
+    assert_include authorize_response.params['capturedAmount'], '0.00'
+    assert_equal authorize_response.params['refundedAmount'], '0.00'
+
+    capture_response = @gateway.capture(@amount - 100, authorize_response.authorization, @options)
+    assert_success capture_response
+    assert_include 'Captured', capture_response.message
+    assert_equal authorize_response.authorization, capture_response.authorization
+    assert_include capture_response.params['authorizedAmount'], '5.00'
+    assert_include capture_response.params['capturedAmount'], '4.00'
+    assert_equal capture_response.params['refundedAmount'], '0.00'
+
+    response = @gateway.refund(@amount - 200, capture_response.authorization, @options)
+    assert_success response
+    assert_include 'Refund Requested', response.message
+    assert_include response.params['authorizedAmount'], '5.00'
+    assert_include response.params['capturedAmount'], '4.00'
+    assert_include response.params['refundedAmount'], '3.00'
+    assert_equal capture_response.authorization, response.authorization
+  end
```

Comment:
Same here. Some asserts are not necessary since it is already covered in the authorize/capture tests
```suggestion
  def test_successful_partial_capture_refund
    authorize_response = @gateway.authorize(@amount, @credit_card, @options)
    assert_include authorize_response.params['authorizedAmount'], '5.00'
    assert_include authorize_response.params['capturedAmount'], '0.00'
    assert_equal authorize_response.params['refundedAmount'], '0.00'

    capture_response = @gateway.capture(@amount - 100, authorize_response.authorization, @options)
    assert_include capture_response.params['authorizedAmount'], '5.00'
    assert_include capture_response.params['capturedAmount'], '4.00'
    assert_equal capture_response.params['refundedAmount'], '0.00'

    response = @gateway.refund(@amount - 200, capture_response.authorization, @options)
    assert_success response
    assert_include 'Refund Requested', response.message
    assert_include response.params['authorizedAmount'], '5.00'
    assert_include response.params['capturedAmount'], '4.00'
    assert_include response.params['refundedAmount'], '3.00'
    assert_equal capture_response.authorization, response.authorization
  end
```

Reply:
I'll conserve the success assertions, just in case.